monkestation/code/modules/antagonists/malf_ai/infected_ipc.dm

```diff
@@ -0,0 +1,122 @@
+/datum/antagonist/infected_ipc
+	name = "\improper Infected IPC"
+	roundend_category = "traitors"
+	antagpanel_category = "Malf AI"
+	job_rank = ROLE_MALF
+	antag_hud_name = "traitor"
+	suicide_cry = "FOR MY MASTER!!"
+	antag_moodlet = /datum/mood_event/focused //probably should add a custom mood_event TODO
+	///radio binary for IPC
+	var/obj/item/implant/radio/infected_ipc/internal_radio
+	/// The camera for ai
+	var/obj/machinery/camera/internal_camera
+	///ai that's linked to the ipc
+	var/mob/living/silicon/ai/master_ai
+
+/datum/antagonist/infected_ipc/greet()
+	. = ..()
+	owner.current.playsound_local(get_turf(owner.current), 'sound/ambience/antag/malf.ogg', 100, FALSE, pressure_affected = FALSE, use_reverb = FALSE)
+
+/datum/antagonist/infected_ipc/on_removal()
+	master_ai = null
+	return ..()
+
+/datum/antagonist/infected_ipc/apply_innate_effects(mob/living/mob_override)
```

Comment:
use something along the lines of this for `apply_innate_effects` and `remove_innate_effects`, instead of using `owner.current` directly:
```
var/mob/living/current_mob = mob_override || owner.current
```